gprofiler/profilers/java.py

```diff
@@ -469,38 +508,54 @@ def __init__(
         java_version_check: bool,
         java_async_profiler_mode: str,
         java_async_profiler_safemode: int,
-        java_safemode: bool,
+        java_safemode: str,
         java_mode: str,
     ):
         assert java_mode == "ap", "Java profiler should not be initialized, wrong java_mode value given"
         super().__init__(frequency, duration, stop_event, storage_dir)
 
-        if java_safemode:
-            assert java_version_check, "Java version checks are mandatory in --java-safemode"
-            assert java_async_profiler_safemode == 127, "Async-profiler safemode must be set to 127 in --java-safemode"
-
         # async-profiler accepts interval between samples (nanoseconds)
         self._interval = int((1 / frequency) * 1000_000_000)
         self._buildids = java_async_profiler_buildids
-        self._version_check = java_version_check
-        if not self._version_check:
+        # simple version check, and
+        self._simple_version_check = java_version_check
+        if not self._simple_version_check:
             logger.warning("Java version checks are disabled")
         self._mode = java_async_profiler_mode
-        self._safemode = java_async_profiler_safemode
+        self._ap_safemode = java_async_profiler_safemode
+        self._init_java_safemode(java_safemode)
         self._saved_mlock: Optional[int] = None
-        self._java_safemode = java_safemode
-        if self._java_safemode:
-            logger.debug("Java safemode enabled")
+        self._should_profile = True
```

Comment:
I changed it because it was more convenient in the code now; and anyway, there's indeed a single instance (of `GProfiler` and all "child" classes), so it's alright.